src/challenges2.js

```diff
@@ -1,23 +1,100 @@
 // Desafio 10
-function techList() {
+// Para resolver o problema de endentação.
+/** Source:  https://eslint.org/docs/2.0.0/rules/guard-for-in */
+function techList(array, myName) {
+  let retorno = [];
+  for (let index in array.sort()) {
+    if ({}.hasOwnProperty.call(array, index)) {
+      retorno[index] = {
+        tech: array[index],
+        name: myName,
+      };
+    }
+  }
+  if (retorno.length === 0) {
+    return 'Vazio!';
+  }
+  return retorno;
   // seu código aqui
 }
 
 // Desafio 11
-function generatePhoneNumber() {
+// Para ajuda nesse desafio, obtive ajuda do Marlon Ramos, ao qual me explicou como criar as funções auxiliares.
+function repeatNumber(numbers, x) {
+  let repetidos = 0;
+  for (let index = 0; index < numbers.length; index += 1) {
+    if (numbers[index] === x) {
+      repetidos += 1;
+    }
+  }
+
+  return repetidos;
+}
+
+function numbereleven(number) {
+  for (let index of number) {
+    if (index < 0 || index > 9 || repeatNumber(number, index) > 2) {
+      return false;
+    }
+  }
+}
+// Para resolução final obtive ajuda do Marlon e também do Regex
+/** Soucer: https://medium.com/@alexandreservian/regex-um-guia-pratico-para-express%C3%B5es-regulares-1ac5fa4dd39f */
+function generatePhoneNumber(numberPhone) {
+  let arraynumber = '';
+  let numbercell = '';
+  let RegExp = /(\d{2})(\d{5})(\d{4})/;
+  if (numberPhone.length !== 11) {
+    arraynumber = `${'Array com tamanho incorreto.'}`;
+  } else if (numbereleven(numberPhone) === false) {
+    arraynumber = `${'não é possível gerar um número de telefone com esses valores'}`;
+  } else {
+    for (let index of numberPhone) {
+      numbercell = numbercell.concat(index);
+    }
+    arraynumber = numbercell.replace(RegExp, '($1) $2-$3');
+  }
+  return arraynumber;
   // seu código aqui
 }
+console.log(generatePhoneNumber([1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1]));
 
 // Desafio 12
-function triangleCheck() {
+// Para completar o desafio, obtive ajuda do Igor Mendes e do Marlon Ramos, Ambos da Turma 11;
+function triangleCheck(lineA, lineB, lineC) {
+  let angl1 = Math.abs(lineA + lineB);
+  let angl2 = Math.abs(lineA - lineB);
+  let value = false;
+
+  if (lineC < angl1 && lineC > angl2) {
+    value = true;
+  } else if (angl1 < lineC && angl2 > lineC) {
```

Comment:
```suggestion
  } else if (angl1 < lineC && angl2 > lineC) {
```
Boa tarde, Julio. Tranquilo? Estava pensando aqui e acho que faria mais sentido você retornar diretamente o 'false', sem a necessidade do else if, pois caso as condições de 'true' não sejam atendidas, ele automaticamente ira retornar 'false'. Mais ou menos desta forma.
if (lineC < angl1 && lineC > angl2) {
  value = true;
  }
  return value;

Faz sentido para você?